packages/block-editor/src/utils/dom.js

```diff
@@ -134,41 +134,50 @@ function isScrollable( element ) {
 	);
 }
 
+export const WITH_OVERFLOW_ELEMENT_BLOCKS = [ 'core/navigation' ];
 /**
- * Returns the rect of the element including all visible nested elements.
- *
- * Visible nested elements, including elements that overflow the parent, are
- * taken into account.
- *
- * This function is useful for calculating the visible area of a block that
- * contains nested elements that overflow the block, e.g. the Navigation block,
- * which can contain overflowing Submenu blocks.
+ * Returns the rect of the element.
  *
+ * Note: the function handles special cases for blocks that contain visible,
+ * nested elements that overflow the block, e.g. the Navigation block,
+ * which can have overflowing submenu blocks. Such blocks are defined in
+ * `WITH_OVERFLOW_ELEMENT_BLOCKS`. For such blocks,
+ * the bounds of the visible children are calculated to determine
+ * the full extent of the block including the visible children that overflow the parent.
  * The returned rect represents the full extent of the element and its visible
  * children, which may extend beyond the viewport.
  *
  * @param {Element} element Element.
  * @return {DOMRect} Bounding client rect of the element and its visible children.
  */
-export function getVisibleElementBounds( element ) {
+export function getElementBounds( element ) {
 	const viewport = element.ownerDocument.defaultView;
 
 	if ( ! viewport ) {
 		return new window.DOMRectReadOnly();
 	}
 
 	let bounds = element.getBoundingClientRect();
-	const stack = [ element ];
-	let currentElement;
-
-	while ( ( currentElement = stack.pop() ) ) {
-		// Children won’t affect bounds unless the element is not scrollable.
-		if ( ! isScrollable( currentElement ) ) {
-			for ( const child of currentElement.children ) {
-				if ( isElementVisible( child ) ) {
-					const childBounds = child.getBoundingClientRect();
-					bounds = rectUnion( bounds, childBounds );
-					stack.push( child );
+	const dataType = element.getAttribute( 'data-type' );
+
+	/*
+	 * Handle special cases for blocks that have overflow elements.
+	 * The bounds of the visible children are calculated to determine the full extent of the block
+	 * including the visible children that overflow the parent.
```

Comment:
```suggestion
	 * For blocks with overflowing elements (like Navigation), include the bounds
	 * of visible children that extend beyond the parent container.
```

Nit: If we're changing the function's docblock description, perhaps a tweak here would also tie this to what has already been read above?